script.js

```diff
@@ -1,4 +1,339 @@
-  return myOutputValue;
+  //Do this if current game mode is setting user name
+  if (currentGameMode == "Waiting for user to enter name") {
+    //set the name
+    userName = input;
+    console.log(currentGameMode);
+    //once the user enters the name, game continues.
+    currentGameMode = "SPR";
+    return (
+      "Hello " +
+      userName +
+      ", Please enter scissors/paper/rock intot he box above"
+    );
+  }
+  //if user types in reverse
+  if (input == "reverse") {
+    currentGameMode = "reversed SPR";
+    console.log(currentGameMode);
+    return "Lets play ;P";
+  }
+  //if user types in unreverse
+  if (input == "unreverse") {
+    currentGameMode = "SPR";
+    console.log(currentGameMode);
+    return "Lets play normally ;P";
+  }
+  if (currentGameMode == "reversed SPR") {
+    //Option is Computer Generated Result of RPS
+    var assignNumberToOption = randomiser();
+    console.log(assignNumberToOption);
+    //Messages
+    var winMessage = "You Win,";
+    var drawMessage = "Draw!";
+    var loseMessage = "You Lose!";
+    var emptyMessage = "Please enter scissors/paper/rock into box";
+    var invalidMessage = "Invalid entry!!!!";
+    var winLossCountMessage =
+      ", Your Win/Loss is " +
+      currentWin +
+      "/" +
+      gameCount +
+      " You have " +
+      currentDraw +
+      " Draws";
+    //scissors lose paper
+    if (input == "scissors" && assignNumberToOption == "paper") {
+      gameCount = gameCount + 1;
+      winLossCountMessage =
+        "Your Win/Loss is " +
+        currentWin +
+        "/" +
+        gameCount +
+        " <br>You have " +
+        currentDraw +
+        " Draws";
+      return loseMessage + "<br>" + winLossCountMessage;
+    }
+    //paper lose rock
+    if (input == "paper" && assignNumberToOption == "rock") {
+      gameCount = gameCount + 1;
+      winLossCountMessage =
+        "Your Win/Loss is " +
+        currentWin +
+        "/" +
+        gameCount +
+        " <br>You have " +
+        currentDraw +
+        " Draws";
+      return loseMessage + "<br>" + winLossCountMessage;
+    }
+    //rock lose scissors
+    if (input == "rock" && assignNumberToOption == "scissors") {
+      gameCount = gameCount + 1;
+      winLossCountMessage =
+        "Your Win/Loss is " +
+        currentWin +
+        "/" +
+        gameCount +
+        " <br>You have " +
+        currentDraw +
+        " Draws";
+      return loseMessage + "<br>" + winLossCountMessage;
+    }
+    //scissors draw
+    if (input == "scissors" && assignNumberToOption == "scissors") {
+      currentDraw = currentDraw + 1;
+      gameCount = gameCount + 1;
+      winLossCountMessage =
+        "Your Win/Loss is " +
+        currentWin +
+        "/" +
+        gameCount +
+        " <br>You have " +
+        currentDraw +
+        " Draws";
+      return drawMessage + "<br>" + winLossCountMessage;
+    }
+    //paper draw
+    if (input == "paper" && assignNumberToOption == "paper") {
```

Comment:
again,  you can make it shorter by directly compare the input and the variable.
 if(input == assignNumberToOption)